Digital 1/Assets/Scripts/ColoredObj.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Digital1
+{
+    /// <summary>
+	/// Base class for colored objects (i.e. player, bullets, enemies).
+	/// </summary>
+    public class ColoredObj : MonoBehaviour
+    {
+        /// <summary>
+        /// List of potential color states of the player/bullet/enemy.
+        /// </summary>
+        public enum ColorState
+        {
+            Neutral,
+            Red,
+            Blue,
+            Yellow
+        }
+
+        //[SerializeField]
+        private Material currentMaterial;
+        //[SerializeField]
+        private MeshRenderer meshRenderer;
+
+        [SerializeField]
+        protected ColorState currentState;
+        public ColorState CurrentState
+        {
+            get { return currentState; }
+        }
+
+        /// <summary>
+        /// Contains list of materials for color changing.
+        /// </summary>
+        [SerializeField]
+        private Material[] colorRef;
+
+        /// <summary>
+		/// Start is called before the first frame update
+		/// </summary>
+		/// <remarks>Known problem where meshRenderer is not initialized here; currently effective workaround is in <see cref="UpdateMaterial"/>.</remarks>
+        void Start()
+        {
+            currentState = ColorState.Neutral;
+            currentMaterial = GetComponent<Material>();
+            meshRenderer = GetComponent<MeshRenderer>();
+			if (meshRenderer == null)
+			{
+                meshRenderer = GetComponentInChildren<MeshRenderer>();
+            }
+        }
+
+        /// <summary>
+        /// Handle Color State switching, then update Material/Mesh.
+        /// </summary>
+        /// <param name="colorIndex"></param>
+        protected void ColorSwitch(int colorIndex)
```

Comment:
This should use the ColorState enum as a parameter instead of an int.

Reply:
I left that from the original pulled code, but I agree, and think that was a change from the revised code I forgot to add.